src/Compilers/Core/MSBuildTask/CanonicalError.cs

```diff
@@ -0,0 +1,413 @@
+                //  foo.cpp(location)
+                // Assume the filename exists, but don't verify it. What else could it be?
+                string location = match.Groups["LOCATION"].Value.Trim();
+                parsedMessage.origin = match.Groups["FILENAME"].Value.Trim();
+
+                // Now, take apart the location. It can be one of these:
+                //      loc:
+                //      (line)
+                //      (line-line)
+                //      (line,col)
+                //      (line,col-col)
+                //      (line,col,len)
+                //      (line,col,line,col)
+                if (location.Length > 0)
+                {
+                    match = lineFromLocation.Match(location);
+                    if (match.Success)
+                    {
+                        parsedMessage.line = ConvertToIntWithDefault(match.Groups["LINE"].Value.Trim());
+                    }
+                    else
+                    {
+                        match = lineLineFromLocation.Match(location);
+                        if (match.Success)
+                        {
+                            parsedMessage.line = ConvertToIntWithDefault(match.Groups["LINE"].Value.Trim());
+                            parsedMessage.endLine = ConvertToIntWithDefault(match.Groups["ENDLINE"].Value.Trim());
+                        }
+                        else
+                        {
+                            match = lineColFromLocation.Match(location);
+                            if (match.Success)
+                            {
+                                parsedMessage.line = ConvertToIntWithDefault(match.Groups["LINE"].Value.Trim());
+                                parsedMessage.column = ConvertToIntWithDefault(match.Groups["COLUMN"].Value.Trim());
+                            }
+                            else
+                            {
+                                match = lineColColFromLocation.Match(location);
+                                if (match.Success)
+                                {
+                                    parsedMessage.line = ConvertToIntWithDefault(match.Groups["LINE"].Value.Trim());
+                                    parsedMessage.column = ConvertToIntWithDefault(match.Groups["COLUMN"].Value.Trim());
+                                    parsedMessage.endColumn = ConvertToIntWithDefault(match.Groups["ENDCOLUMN"].Value.Trim());
+                                }
+                                else
+                                {
+                                    match = lineColLineColFromLocation.Match(location);
+                                    if (match.Success)
+                                    {
+                                        parsedMessage.line = ConvertToIntWithDefault(match.Groups["LINE"].Value.Trim());
+                                        parsedMessage.column = ConvertToIntWithDefault(match.Groups["COLUMN"].Value.Trim());
```

Comment:
Maybe ConvertToIntWithDefault should do the trim instead of all of the call sites.